UglyTicTacToe/UglyTicTacToe/Game.aspx.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+
+namespace UglyTicTacToe
+{
+    public enum Player
+    {
+        None,
+        X,
+        O
+    }
+    public partial class Game : Base
+    {
+       
+
+        private static string TurnKey = "turn";
+        public bool Turn
+        {
+            get
+            {
+                if (ViewState[TurnKey] == null)
+                {
+                    return true;
+                }
+                else
+                {
+                    return (bool)ViewState[TurnKey];
+                }
+            }
+            set
+            {
+                ViewState[TurnKey] = value;
+            }
+        }
+      
+        protected void Page_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        protected void TakeTurn(Button button, int row, int col)
+        {
+            Board.NextTurn();
+            if (Turn == true)
+            {
+                button.Text = "X";
+                Board.GetBoard()[row, col] = Player.X;
+                Turn = false;
+
+                if (Win(Player.X))
+                {
+                    Winner.Text = "Congratulations Player X, You Won!";
+                    return;
+                }
```

Comment:
The code in this `if` statement block could be put into its own method. Then you could call that method here and in the `else` statement block, passing in either `Player.X` or `Player.O`.